### DISCUSSION TITLE
Clearing State not working as expected

### DISCUSSION THREAD

**Discussion by stkmr** [xyflow/xyflow#4210] [Q&A]
Hi there,

i'm pretty confident i'm just doing it wrong, but im trying basically the following: 

1) receive some data, paint a flow
2) replace that flow with some different data

 ```
               <ReactFlow
                    nodes={GetNodesState}
                    edges={GetEdgesState}
                    onNodesChange={onNodesChange}
                    onEdgesChange={onEdgesChange}
                    onInit={onInit}
                    edgesUpdatable={false}
                    edgesFocusable={false}
                    nodesDraggable={false}
                    nodesConnectable={false}

                    fitView
                ></ReactFlow>
```

and

 ```
   const { getNodes, setNodes, getEdges, setEdges, fitView } = useReactFlow();
    const { getLayoutedElements } = useLayoutedElements();
    const [GetNodesState, SetNodesState, onNodesChange] = useNodesState([]);
    const [GetEdgesState, SetEdgesState, onEdgesChange] = useEdgesState([]);
```

when trying to make those calls within an function, they dont have any effect:

  ```
      setEdges([]);
        SetEdgesState([])

        console.log("--------------------------")
        console.log(getEdges());
        console.log(GetEdgesState);
        console.log("--------------------------")
```

I would expect to have no edges in the state afterwards, but it still remains the state before. What I'm doing wrong here?

Would really appreciate your help!

Thanks in advance!

**Comment by bcakmakoglu** [comment]
Pretty sure they *do* have an effect but you're treating `setEdges` as if it was instantly applied, which it isn't.
When you call the internal `setEdges` function, what happens is that a set of *changes* are generated, which are propagated to you through `onEdgesChange`.
By the time you call `getEdges` (or `GetEdgesState`) this change has not been applied yet (the same goes for `SetEdgesState`, neither of these are applied by the time you are calling the logs to check if the array is empty or not).

**Comment by stkmr** [comment]
Thanks a lot! Figured it out, you exactly described my issue, got it working now!
